File: Exercism/go/isogram/isogram.go
package isogram

import (
	"strings"
)

func IsIsogram(input string) bool {
	var letters = make(map[rune]bool)

	for _, c := range strings.ToLower(input) {
		if c == ' ' || c == '-' {
			continue
		}

		if _, contains := letters[c]; contains {
			return false
		}

		letters[c] = true
	}

	return true
}

File: Exercism/cpp/isogram/isogram.cpp
#include "isogram.h"

#include <vector>

using std::string;
using std::vector;

namespace isogram {

bool is_isogram(string input)
{
    vector<bool> letters(26, false);

    for (string::size_type i = 0; i < input.length(); ++i) {
        if (!std::isalpha(input[i]))
            continue;

        if (letters[std::tolower(input[i]) - 97])
            return false;

        letters[std::tolower(input[i]) - 97] = true;
    }

    return true;
}

} // namespace isogram
